### PR TITLE
PMP: Document overload of angle_and_area_smoothing()

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Add a function documentation

## Release Management

* Affected package(s): Polygon_mesh_processing
* Issue(s) solved (if any): fix #7487 
* Feature/Small Feature (if any):
* Link to compiled documentation:  [link](https://cgal.geometryfactory.com/CGAL/Manual_doxygen_test/CGAL-5.6-Ic-264/output2/Polygon_mesh_processing/group__PMP__meshing__grp.html#ga6e971b85a933e194198218130ba98c0b) into doc testsuite
* License and copyright ownership:

### CODE REVIEW

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h [hunk 1]
```diff
@@ -329,13 +329,33 @@ void angle_and_area_smoothing(const FaceRange& faces,
   }
 }
 
-///\cond SKIP_IN_MANUAL
+/*!
+* \ingroup PMP_meshing_grp
+*
+* \short smooths a polygon mesh.
```
Comment (albert-github) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h:335]: Better to use the equivalent `\brief`:
- only in this file the `\short` is used (in the past as well) but nowhere else in CGAL

**Comment by afabri** [comment]
@lrineau the test is failing because the PR targets 5.4, if I understood @sloriot correctly.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h (hunk 1 above)
Comment (afabri) [reply Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h]: fixed. Thanks for pointing out.

**Comment by lrineau** [comment]
> @lrineau the test is failing because the PR targets 5.5, if I understood @sloriot correctly.

@sloriot: Should we test the older branches with the older Doxygen version?

**Comment by sloriot** [comment]
For the CI it is possible but not in the testsuite.

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h [hunk 2]
```diff
@@ -329,13 +329,34 @@ void angle_and_area_smoothing(const FaceRange& faces,
...
+* might change the combinatorial information, unless specified otherwise. It is also possible
+* to make the smoothing algorithm "safer" by rejecting moves that, when applied, would worsen the
+* quality of the mesh, e.g. that would decrease the value of the smallest angle around a vertex or
+* create self-intersections.
+*
+* Optionally, the points are reprojected after each iteration.
+*
+* The overload of this function which takes a face range as
+* additonal parameter documents the template
+* parameters, parameters, and named parameters.
```
Comment (MaelRL) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/angle_and_area_smoothing.h:350]:
```suggestion
* See the overload which takes a face range as additonal parameter for a comprehensive description
* of the parameters. 
```

**Comment by afabri** [comment]
> For the CI it is possible but not in the testsuite.

We discussed this yesterday and the conclusion was not to waste time on it.

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.6-Ic-264

**Pull request merged**